File: src/BuilderPattern/BuilderPatternMain.java
package BuilderPattern;

import BuilderPattern.CarBuilder.Car;
import BuilderPattern.CarBuilder.CarBuilder;
import BuilderPattern.CarBuilder.CarDirector;
import BuilderPattern.CarBuilder.ICarBuilder;

import java.awt.Color;

public class BuilderPatternMain {
    public static void main(String... args) {

//        Car car = new Car("Audi", "A6", 4, Color.BLACK); // unable to initialize Car like this, since the Constructor is protected

        // initialize the director and CarBuilder
        CarDirector director = new CarDirector();
        ICarBuilder builder = new CarBuilder();

        //build default car
        director.buildDefaultCar(builder);
        Car defaultCar = builder.build();

        // build white Smart
        director.buildWhiteSmart(builder);
        Car whiteSmart = builder.build();

        // build other custom Cars
        Car whitePolo = builder.brand("VW").model("Polo").color(Color.WHITE).build();
        Car blackFordMustang = builder.brand("Ford").model("Mustang").color(Color.BLACK).amountOfDoors(2).build();
    }

}

File: BuilderPattern/CSharp/BuilderPatternMain.cs
using BuilderPattern.NSCarBuilder;
using System;
using System.Collections.Generic;
using System.Drawing;
using System.Linq;
using System.Text;
using System.Threading.Tasks;

namespace BuilderPattern {

    public class BuilderPatternMain {
        public static void Main(string[] args) {

            //Car car = new Car("Audi", "A6", 4, Color.BLACK); // unable to initialize Car like this, since the Constructor is protected

            // initialize the director and CarBuilder
            CarDirector director = new CarDirector();
            ICarBuilder builder = new CarBuilder();

            //build default car
            director.BuildDefaultCar(builder);
            Car defaultCar = builder.Build();

            // build white Smart
            director.BuildWhiteSmart(builder);
            Car whiteSmart = builder.Build();

            // build other custom Cars
            Car whitePolo = builder.Brand("VW").Model("Polo").Color(Color.White).Build();
            Car blackFordMustang = builder.Brand("Ford").Model("Mustang").Color(Color.Black).AmountOfDoors(2).Build();
        }

    }
}
